collect_app/src/main/java/org/odk/collect/android/http/injection/HttpInterfaceModule.java

```diff
@@ -0,0 +1,27 @@
+package org.odk.collect.android.http.injection;
+
+import org.odk.collect.android.http.CollectServerClient;
+import org.odk.collect.android.http.HttpClientConnection;
+import org.odk.collect.android.http.OpenRosaHttpInterface;
+
+import javax.inject.Singleton;
+
+import dagger.Module;
+import dagger.Provides;
+
+@Module
+public class HttpInterfaceModule {
```

Comment:
@xsteelej I am thinking we should hold off on creating this `HttpInterfaceModule` for now. It is the right approach to properly organize all the HTTP related dependencies in one component but what's happening now is that if a target class wants to `HttpComponent` and the `ApplicationComponent` as well then there will be an issue because two components can be injected at once. What we would have to do is simply have an application component that's providing the other sub/child components i.e we will have to make proper usage of the subcomponent & component dependencies features of Dagger. With that said, I think we could probably just leave that refactor for another time and place all these `provides` methods in the `ApplicationModule` to be safe for now.

Reply:
Makes sense, i'll move all the @provides out of HttpInterfaceModule and into ApplicationModule.